### ISSUE TITLE
Load faster

### ISSUE THREAD

**Issue by alfredplpl** [lucidrains/denoising-diffusion-pytorch#36]
The code seems that the image loading method is slow.
For example, I think the code should load learning images in parallel.
Current:
https://github.com/lucidrains/denoising-diffusion-pytorch/blob/f900f40f14a7f24b8ef2145b2d2af221b46b4ecc/denoising_diffusion_pytorch/denoising_diffusion_pytorch.py#L601

After:
`self.dl = cycle(data.DataLoader(self.ds, batch_size = train_batch_size, num_workers=os.cpu_count(), shuffle=True, pin_memory=True))`

May I make an pull request?

**Comment by alfredplpl** [comment]
Thank you for your coding.
